pbcommand/utils.py

```diff
@@ -245,4 +249,47 @@ def ignored(*exceptions):
     try:
         yield
     except exceptions:
-        pass
\ No newline at end of file
+        pass
+
+
+def get_dataset_metadata(path):
+    """
+    Returns DataSetMeta data or raises ValueError, KeyError
+
+    :param path:
+    :return:
+    """
+    f = ET.parse(path).getroot().attrib
+    mt = f['MetaType']
+    uuid = f['UniqueId']
+    if mt in FileTypes.ALL_DATASET_TYPES().keys():
+        return DataSetMetaData(uuid, mt)
+    else:
+        raise ValueError("Unsupported dataset type '{t}'".format(t=mt))
+
+
+def get_dataset_metadata_or_none(path):
+    """
+    Returns DataSetMeta data, else None
+
+    :param path:
+    :return:
+    """
+    try:
+        return get_dataset_metadata(path)
+    except:
```

Comment:
Bare excepts swallow KeyboardInterrupt along with exceptions raised by the code.